internal/tools/k8s/namespaces.go

```diff
@@ -23,16 +23,27 @@ func (kc *KubeClient) CreateNamespacesIfDoesNotExist(namespaceNames []string) ([
 	return namespaces, nil
 }
 
-// CreateNamespaceIfDoesNotExist creates a kubernetes namespace if it doesn't
-// exist already.
-func (kc *KubeClient) CreateNamespaceIfDoesNotExist(namespaceName string) (*corev1.Namespace, error) {
-	ns, err := kc.Clientset.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
-	if err == nil {
-		return ns, nil
+// CreateOrUpdateNamespace creates or update a namespace
+func (kc *KubeClient) CreateOrUpdateNamespace(namespaceName string) (*corev1.Namespace, error) {
+	_, err := kc.Clientset.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return nil, err
 	}
 
-	nsSpec := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
-	return kc.Clientset.CoreV1().Namespaces().Create(nsSpec)
+	nsSpec := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+			Labels: map[string]string{
+				"name": namespaceName,
```

Comment:
yes, then we can use in the network policies to filter, otherwise, we cannot filter by namespaces

this can be anything, for now just the name is enough